### PR TITLE
Added userprofile, currently displays first filteredUser when searchi…

### PR DESCRIPTION
**Pull request by glarence**
…ng as you type

### CODE REVIEW

File: basicapp/src/UserList.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function UserList (props) {
+    return props.users.map(user => {
```
Comment (zhangxian97) [inline basicapp/src/UserList.js:2]: the return value of this function is an array, not a legal component.

File: basicapp/src/App.js [hunk 2]
```diff
@@ -32,55 +34,32 @@ function App() {
   const [searchQuery, setSearchQuery] = useState(query || '');
   const filteredUsers = filterUsernames(users, searchQuery);
```
Comment (zhangxian97) [inline basicapp/src/App.js:35]: this should be probably a state and only set when query changes

File: basicapp/src/App.js [hunk 3]
```diff
@@ -32,55 +34,32 @@ function App() {
   const [searchQuery, setSearchQuery] = useState(query || '');
```
Comment (zhangxian97) [inline basicapp/src/App.js:34]: query should not come from url. it should be from the search bar input?

**Comment by glarence** [comment]
cleaned up code

File: aws_lambda_devguide_nodejs-apig/function/index.js [hunk 4]
```diff
@@ -4,15 +4,17 @@ const lambda = new AWS.Lambda()
 
 // Handler
 exports.handler = async function(event, context) {
-  console.log('## ENVIRONMENT VARIABLES: ' + serialize(process.env))
+  /*console.log('## ENVIRONMENT VARIABLES: ' + serialize(process.env))
```
Comment (xianzhangvs) [inline aws_lambda_devguide_nodejs-apig/function/index.js:7]: remove this?

File: react_withaws_ssrlambda_sample/simple-ssr/src/App.js [hunk 5]
```diff
@@ -32,19 +32,19 @@ const App = ({ isSSR, ssrData }) => {
...
       <div>
           <SearchBar />
       </div>
 
       </header>
-      </div>*/
+      </div>
 
-      <div>
+      /*<div>
```
Comment (xianzhangvs) [inline react_withaws_ssrlambda_sample/simple-ssr/src/App.js:44]: remove this?

File: react_withaws_ssrlambda_sample/simple-ssr/src/SSRApp.js [hunk 6]
```diff
@@ -14,19 +14,19 @@ const SSRApp = ({ data }) => {
...
           <SearchBar />
       </div>
 
       </header>
-      </div>*/
+      </div>
 
-      <div>
-        <h1>Hello Universe!</h1>
+      /*<div>
```
Comment (xianzhangvs) [inline react_withaws_ssrlambda_sample/simple-ssr/src/SSRApp.js:26]: same remvoe this.

File: aws_lambda_devguide_nodejs-apig/function/index.js (hunk 4 above)
Comment (glarence) [reply aws_lambda_devguide_nodejs-apig/function/index.js:7]: I think that's default code comment from aws nodejs apig sample, not mine

Comment (xianzhangvs) [reply aws_lambda_devguide_nodejs-apig/function/index.js:7]: yeah as long as you don't need it, remove it

**Review by glarence — COMMENTED** [review]
userprofile functionality when searching for user, also updated AWS deployment

**Pull request merged**